### PR TITLE
add oss bucket policy implementation

### PR DESCRIPTION
**Pull request by coderall**

### CODE REVIEW

File: GNUmakefile [hunk 1]
```diff
@@ -84,8 +84,10 @@ windows:
...
 	tar czvf bin/terraform-provider-alicloud_windows-amd64.tgz bin/terraform-provider-alicloud.exe
 	rm -rf bin/terraform-provider-alicloud.exe
+	tar -xvf bin/terraform-provider-alicloud_windows-amd64.tgz && mv bin/terraform-provider-alicloud $(shell dirname `which terraform`)
 
 linux:
 	GOOS=linux GOARCH=amd64 go build -o bin/terraform-provider-alicloud
 	tar czvf bin/terraform-provider-alicloud_linux-amd64.tgz bin/terraform-provider-alicloud
-	rm -rf bin/terraform-provider-alicloud
\ No newline at end of file
+	rm -rf bin/terraform-provider-alicloud
```
Comment (xiaozhu36) [inline GNUmakefile]: I don't suggest to update the linux and windows. You can add linuxcopy and windowscopy to fix it.

File: alicloud/connectivity/client.go [hunk 2]
```diff
@@ -31,10 +50,8 @@ import (
 	"github.com/aliyun/alibaba-cloud-sdk-go/services/sts"
 	"github.com/aliyun/alibaba-cloud-sdk-go/services/vpc"
 	"github.com/aliyun/aliyun-datahub-sdk-go/datahub"
-	"github.com/aliyun/aliyun-log-go-sdk"
 	"github.com/aliyun/aliyun-oss-go-sdk/oss"
 	"github.com/aliyun/aliyun-tablestore-go-sdk/tablestore"
-	"github.com/aliyun/fc-go-sdk"
 	"github.com/denverdino/aliyungo/cdn"
 	"github.com/denverdino/aliyungo/cs"
 	"github.com/dxh031/ali_mns"
```
Comment (xiaozhu36) [inline alicloud/connectivity/client.go]: I don't think this file should be modified.

File: alicloud/data_source_alicloud_oss_buckets.go [hunk 3]
```diff
@@ -389,6 +396,27 @@ func bucketsDescriptionAttributes(d *schema.ResourceData, buckets []oss.BucketPr
...
+			if err != nil {
+				return err
+			}
+			policy = string(rawData)
+		} else {
+			return err
+		}
+		mapping["policy"] = policy
+
+		log.Printf("[DEBUG] alicloud_oss_buckets - adding bucket mapping: %v", mapping)
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_oss_buckets.go]: Please remove the line 418

File: alicloud/resource_alicloud_oss_bucket.go [hunk 4]
```diff
@@ -119,12 +120,12 @@ func resourceAlicloudOssBucket() *schema.Resource {
 			"referer_config": {
 				Type:     schema.TypeList,
 				Optional: true,
+				//Computed: true,
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_oss_bucket.go]: Please remove the useless annotation.

File: alicloud/resource_alicloud_oss_bucket.go [hunk 5]
```diff
@@ -414,6 +419,26 @@ func resourceAlicloudOssBucketRead(d *schema.ResourceData, meta interface{}) err
...
+	raw, err = client.WithOssClient(func(ossClient *oss.Client) (interface{}, error) {
+		params := map[string]interface{}{}
+		params["policy"] = nil
+		return ossClient.Conn.Do("GET", d.Id(), "", params, nil, nil, 0, nil)
+	})
+
+	rawResp := raw.(*oss.Response)
+	defer rawResp.Body.Close()
+
+	if err == nil {
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_oss_bucket.go]: If there is an error, it should be throw.

File: alicloud/data_source_alicloud_oss_buckets.go [hunk 6]
```diff
@@ -190,6 +192,11 @@ func dataSourceAlicloudOssBuckets() *schema.Resource {
 								},
 							},
 						},
+
+						"policy": {
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_oss_buckets.go]: this parameter should be added into website/docs

File: GNUmakefile (hunk 1 above)
Comment (coderall) [reply GNUmakefile]: OK

File: alicloud/data_source_alicloud_oss_buckets.go (hunk 6 above)
Comment (coderall) [reply alicloud/data_source_alicloud_oss_buckets.go]: OK

File: alicloud/resource_alicloud_oss_bucket.go (hunk 4 above)
Comment (coderall) [reply alicloud/resource_alicloud_oss_bucket.go]: done

File: alicloud/resource_alicloud_oss_bucket.go (hunk 5 above)
Comment (coderall) [reply alicloud/resource_alicloud_oss_bucket.go]: done

File: website/docs/r/oss_bucket.html.markdown [hunk 7]
```diff
@@ -96,18 +96,38 @@ resource "alicloud_oss_bucket" "bucket-lifecycle" {
...
-* `lifecycle_rule` - (Optional) A list configurations of [object lifecycle management](https://www.alibabacloud.com/help/doc-detail/31904.htm) (documented below). The items of rules are no more than 1000 for every OSS bucket.
+* `core_rule` - (Optional) A rule of [Cross-Origin Resource Sharing](https://www.alibabacloud.com/help/doc-detail/31903.htm) (documented below). The items of core rule are no more than 10 for every OSS bucket.
+* `website` - (Optional) A website object(documented below).
+* `logging` - (Optional) A Settings of [bucket logging](https://www.alibabacloud.com/help/doc-detail/31900.htm) (documented below).
+* `logging_isenable` - (Optional) The flag of using logging enable container. Defaults true.
+* `referer_config` - (Optional) The configuration of [referer](https://www.alibabacloud.com/help/doc-detail/31901.htm) (documented below).
+* `lifecycle_rule` - (Optional) A configuration of [object lifecycle management](https://www.alibabacloud.com/help/doc-detail/31904.htm) (documented below).
+* `policy` - (Optional) Json format text of bucket policy [bucket policy management](https://www.alibabacloud.com/help/doc-detail/100680.htm) (documented below).
```
Comment (xiaozhu36) [inline website/docs/r/oss_bucket.html.markdown]: This should add a tag for policy `policy - (Optional, Available in 1.41.0)`

Comment (coderall) [reply website/docs/r/oss_bucket.html.markdown]: done

**Pull request merged**